src/glossary.rst

```diff
@@ -1381,10 +1410,37 @@ Glossary
          foo => bar  # means foo:succeeded => bar
 
 
+   output completion
+   output completion condition
+      A task's outputs are *complete* if its *output completion condition* 
+      is satisfied.
+
+      The completion condition can be defined by the user in
+      :cylc:conf:`flow.cylc[runtime][<namespace>]completion`,
+      otherwise it is automatically generated according to the rules:
+
+      - All :term:`required outputs <required output>` (referenced in the graph)
+        must be completed.
+      - Or, if success is optional, then the outputs are complete if it fails.
+      - Or, if submission is optional, then the outputs are complete if it
+        submit-fails.
+      - Or, if expiry is optional, then the outputs are complete if it expires.
+
+      Tasks that achieve a :term:`final status` with complete outputs have done
+      their job, allowing the workflow to move on.
+
+      Tasks that achieve a final status with incomplete outputs are retained in
+      :term:`n=0 <n-window>` pending user intervention, and will :term:`stall`
+      the workflow.
+
+
+   prerequisite
    dependence
    dependency
       Dependencies in the :term:`graph` show how :term:`tasks <task>` depend on
       some combination of the :term:`outputs <task output>` of other tasks.
+      These dependencies are also called the prerequisites of the downstream.
+      task.
```

Comment:
```suggestion
      These dependencies are also called the prerequisites of the downstream
      task.
```

Or possibly get rid of this, as I find the wording a little confusing, and I think the definition of prerequisite is already clear.

```suggestion
```